trts/src/lib.rs

```diff
@@ -4,37 +4,29 @@
 #![no_std]
 #![deny(missing_docs, missing_debug_implementations)]
 
-use core::{ffi::c_void, mem};
+use core::ffi::c_void;
 use mc_sgx_trts_sys::{sgx_is_outside_enclave, sgx_is_within_enclave};
 
 /// Behavior for determining where memory values are at.
 pub trait EnclaveMemory<T> {
     /// Is the item fully within the enclave's memory space?
-    ///
-    /// # Arguments
-    /// - `item` - The item to see if it's fully inside of the enclave's memory
-    ///   space
-    fn is_within_enclave(item: &T) -> bool;
+    fn is_within_enclave(&self) -> bool;
 
     /// Is the item fully outside of the enclave's memory space?
-    ///
-    /// # Arguments
-    /// - `item` - The item to see if it's fully outside of the enclave's memory
-    ///   space
-    fn is_outside_enclave(item: &T) -> bool;
+    fn is_outside_enclave(&self) -> bool;
 }
 
-impl<T: Sized> EnclaveMemory<T> for T {
-    fn is_within_enclave(item: &T) -> bool {
-        let start = item as *const _ as *const c_void;
-        let size = mem::size_of::<T>();
```

Comment:
I think it's probably best if these are bare functions that use [`core::mem::size_of_val()`](https://doc.rust-lang.org/src/core/mem/mod.rs.html#342-345) instead of size_of()...

Reply:
This went away and currently it only supports vecs and slices.  

I think a trait for vectors is kind of nice:
```rust
        if !data.is_within_enclave() {
```
But I could just be latching on to a new hammer